truffleruby/src/main/ruby/post-boot/post-boot.rb

```diff
@@ -16,6 +16,27 @@
     require 'rubygems'
   rescue LoadError
   else
+    unless Gem.dir.include?(Truffle::Boot.ruby_home)
+      bad_gem_home = false
+
+      # rbenv does not set GEM_HOME
+      # rbenv-gemset has to be installed which does set GEM_HOME, it's in the subdir of Truffle::Boot.ruby_home
+      # rbenv/versions/<ruby>/gemsets
+      bad_gem_home ||= Gem.dir.include?('rbenv/versions') && !Gem.dir.include?('rbenv/versions/truffleruby')
+
+      # rvm stores gems at .rvm/gems/<ruby>@<gemset-name>
+      bad_gem_home ||= Gem.dir.include?('rvm/gems') && !Gem.dir.include?('rvm/gems/truffleruby')
+
+      # chruby stores gem in ~/.gem/<ruby>/<version>
+      bad_gem_home ||= Gem.dir.include?('.gem') && !Gem.dir.include?('.gems/truffleruby')
```

Comment:
The condition you are suggesting will catch any GEM_HOME change, the old one catches only different GEM_HOME pointing to a gem-home of another ruby in a ruby manager. That is the missing preciseness in the new condition I am referring too. Gem path can be anything, it cannot be easily checked that it's unintentional change and that the path belongs to other ruby. Anyway we need to only detect the common case of misconfigured a ruby manager. We ignore other changes. This will be deleted when truffleruby can be added to ruby managers.